### PR TITLE
UI Widgets: test modules refactoring

### PR DESCRIPTION
**Pull request by ksercs**
1) Old type modules replaced with new modules with callbacks

before: 
```
QUnit.module(moduleName, moduleConfig);
QUnit.test(testName, testFunction);
```
after: 
```
QUnit.module(moduleName, moduleConfig, () => {
    QUnit.test(testName, testFunction);
});
```
2) Module hooks replaced with moduleConfig object

before: 
```
QUnit.module(moduleName, (hooks) => {
    hooks.beforeEach(() => {
        ...
    }),

    hooks.afterEach(() => {
        ...
    }),

    QUnit.test(testName, testFunction);
```
after: 
```
QUnit.module(moduleName, moduleConfig, () => {
    QUnit.test(testName, testFunction);
});
```
3) Anonymous wrapper functions removed (it's useless now)

before: 
```
(function() {
    QUnit.module(moduleName, moduleConfig);
    QUnit.test(testName, testFunction);
})(moduleName);
```
after: 
```
QUnit.module(moduleName, moduleConfig, () => {
    QUnit.test(testName, testFunction);
});
```
4)  - ESLint warnings fixed
     - extra empty lines/empty modules/empty callbacks removed

Result: 
1) All modules created in the same way:

```
QUnit.module(moduleName, moduleConfig, () => {
    QUnit.test(testName, testFunction);
});
``` 
2) Whole module/ module tests/ module config can be collapsed => convenient navigation
3) All files are refactored

### CODE REVIEW

File: testing/tests/DevExpress.ui.widgets.editors/validationSummary.tests.js [hunk 1]
```diff
@@ -38,433 +38,436 @@ QUnit.testStart(function() {
...
 
-        // assert
-        var itemElements = this.fixture.$summaryContainer.find('.dx-validationsummary-item');
-        assert.equal(itemElements.length, 1, 'Single item element should be rendered');
-        itemElements.trigger('click');
-        assert.ok(validator.focus.calledOnce, 'Validator should be focused');
-    });
-})('General');
 
+    (function() {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validationSummary.tests.js]: Something wrong here...

File: testing/tests/DevExpress.ui.widgets.editors/validationSummary.tests.js [hunk 2]
```diff
@@ -38,433 +38,436 @@ QUnit.testStart(function() {
...
 
-        // fake
-        summary._groupValidationHandler = sinon.spy();
-        // act
-        summary.option('validationGroup', group);
-        ValidationEngine.getGroupConfig(undefined).validate();
-        // assert
-        assert.ok(summary._groupValidationHandler.notCalled, 'Handler should not be called');
 
+    (function() {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validationSummary.tests.js]: Before that, this function ended further.

File: testing/tests/DevExpress.ui.widgets.editors/validatorIntegration.tests.js [hunk 3]
```diff
@@ -222,35 +252,6 @@ var Fixture = Class.inherit({
...
-            }]
-        }, false);
-
-        this.fixture.validator.reset();
-
-        assert.notOk(validationCallback.called, 'validationCallback should not be called');
-    });
-})('Regression');
+    })('Regression');
+});
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets.editors/validatorIntegration.tests.js:253]: It looks like [this function](https://github.com/DevExpress/DevExtreme/pull/11270/files#diff-b13f202a4de538a52b9905e0430824f6R55) is never called now.

File: testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js [hunk 4]
```diff
@@ -140,7 +140,7 @@ QUnit.module('create multiple actions', (hooks) => {
 
         assert.equal($fabMainElement.attr('title'), 'Arrow down', 'hint by first action option');
     });
-});
+}, () => {});
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js]: Do we need this?

File: testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js [hunk 5]
```diff
@@ -246,7 +246,7 @@ QUnit.module('modify global action button config', (hooks) => {
         assert.equal($fabMainContent.offset().left, $(window).width() - fabDimensions, 'multiple actions - correct position left after config change');
 
     });
-});
+}, () => {});
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js]: Do we need this?

File: testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js [hunk 6]
```diff
@@ -293,7 +293,7 @@ QUnit.module('add or remove action buttons', (hooks) => {
 
         assert.equal($fabMainContent.offset().top, $(window).height() - fabMainOffsetY - $fabMainContent.height(), 'use dafault position');
     });
-});
+}, () => {});
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js]: Do we need this?

File: testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js [hunk 7]
```diff
@@ -362,7 +362,7 @@ QUnit.module('check action buttons position', (hooks) => {
         assert.equal($fabWrapper.eq(0).css('position'), expectedPosition, 'first action has the same position with main fab');
         assert.equal($fabWrapper.eq(1).css('position'), expectedPosition, 'second action has the same position with main fab');
     });
-});
+}, () => {});
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js]: Do we need this?

File: testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js [hunk 8]
```diff
@@ -400,7 +400,7 @@ QUnit.module('check action buttons click args', (hooks) => {
         $fabMainContent.trigger('dxclick');
         $fabContent.eq(2).trigger('dxclick');
     });
-});
+}, () => {});
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js]: Do we need this?

File: testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js [hunk 9]
```diff
@@ -486,7 +486,7 @@ QUnit.module('add label option', (hooks) => {
         assert.ok($(FAB_SELECTOR).find('.dx-overlay-content').eq(1).hasClass(FAB_CONTENT_REVERSE_CLASS), 'first SDA has label on the right');
         assert.ok($(FAB_SELECTOR).find('.dx-overlay-content').eq(2).hasClass(FAB_CONTENT_REVERSE_CLASS), 'second SDA has label on the right');
     });
-});
+}, () => {});
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js]: Do we need this?

File: testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js [hunk 10]
```diff
@@ -604,7 +604,7 @@ QUnit.module('add visible option', (hooks) => {
 
         assert.equal($(FAB_INVISIBLE_SELECTOR).length, 3, 'all actions are invisible');
     });
-});
+}, () => {});
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js]: Do we need this?

File: testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js [hunk 11]
```diff
@@ -705,7 +705,7 @@ QUnit.module('add shading option', (hooks) => {
         firstSDA.dispose();
         secondSDA.dispose();
     });
-});
+}, () => {});
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js]: Do we need this?

File: testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js [hunk 12]
```diff
@@ -783,7 +783,7 @@ QUnit.module('add direction option', (hooks) => {
...
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js]: Do we need this?

File: testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js [hunk 13]
```diff
@@ -837,7 +837,7 @@ QUnit.module('add index option', (hooks) => {
...
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js]: Do we need this?

File: testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js [hunk 14]
```diff
@@ -912,4 +912,5 @@ QUnit.module('check action buttons events', (hooks) => {
         assert.ok(contentReadyTwoArgs[0].actionElement.hasClass('dx-overlay'), 'right second SDA content ready actionElement in args');
         assert.equal($(contentReadyTwoArgs[0].element).attr('id'), 'fab-two', 'right second SDA content ready element in args');
     });
-});
+}, () => {});
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/speedDialAction.tests.js]: Do we need this?

File: testing/tests/DevExpress.ui.widgets/tileView.markup.tests.js [hunk 15]
```diff
@@ -105,67 +105,68 @@ QUnit.module('rendering', {
...
-    var $items = element.find(TILEVIEW_ITEM_SELECTOR);
-
-    assert.equal($items.eq(0).text(), 'Text is: item1');
-});
-
 
-$.each(configs, function(direction, config) {
 
-    var getPosition = getPositionCreator(config);
+    $.each(configs, function(direction, config) {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/tileView.markup.tests.js]: Looks like we do not need the indentation for this block.

File: testing/tests/DevExpress.ui.widgets/tileView.markup.tests.js [hunk 16]
```diff
@@ -105,67 +105,68 @@ QUnit.module('rendering', {
...
+        var getPosition = getPositionCreator(config);
 
-    QUnit.test('items positions should be correct', function(assert) {
-        var element = this.element.dxTileView({
-            direction: direction,
-            height: 200,
-            width: 200,
-            items: prepareItems(items, config)
-        });
+        QUnit.module('rendering ' + direction, {
```
Comment (San4es) [inline testing/tests/DevExpress.ui.widgets/tileView.markup.tests.js]: Now, this module has become nested to the `rendering` module. Is it conceived?

**Review by IgnatovDan — APPROVED** [review]
My check was a lightweight one: number of tests before/after and few files.

**Review by San4es — APPROVED** [review]

**Review by vladkovl — APPROVED** [review]

**Review by alexander-kotov-dx — APPROVED** [review]

**Review by kruglikovstepan — APPROVED** [review]

**Review by DokaRus — APPROVED** [review]

**Pull request merged**